x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/EmailService.java

```diff
@@ -174,10 +192,16 @@ public EmailService(Settings settings, @Nullable CryptoService cryptoService, SS
         clusterSettings.addAffixUpdateConsumer(SETTING_SMTP_LOCAL_PORT, (s, o) -> {}, (s, o) -> {});
         clusterSettings.addAffixUpdateConsumer(SETTING_SMTP_SEND_PARTIAL, (s, o) -> {}, (s, o) -> {});
         clusterSettings.addAffixUpdateConsumer(SETTING_SMTP_WAIT_ON_QUIT, (s, o) -> {}, (s, o) -> {});
+        this.allowedDomains = new HashSet<>(SETTING_DOMAIN_ALLOWLIST.get(settings));
+        clusterSettings.addSettingsUpdateConsumer(SETTING_DOMAIN_ALLOWLIST, (s) -> {});
```

Comment:
Question for my own curiosity: What is the benefit of this consumer if it doesn't perform any actions? It looks like we mark the setting as dynamic, but we don't actually update it anywhere?

Reply:
According to a comment above, it ensures that the setting is logged when it's changed, but I didn't test to ensure that (just followed the other examples)

Reply:
I think it needs to update `allowedDomains` -- https://github.com/elastic/elasticsearch/pull/90426